packages/quantic/force-app/solutionExamples/main/lwc/resultPostToFeed/resultPostToFeed.js

```diff
@@ -0,0 +1,160 @@
+ * that can be caught by Aura wrapper to insert the result using the Salesforce Quick Action API.
+ */
+export default class ResultPostToFeed extends LightningElement {
+  actionName = 'FeedItem.TextPost';
+  labels = {
+    postToFeed: postToFeed,
+    errorInsertingTheResultInChatter: 'error',
+    errorWithQuickAction: I18nUtils.format(
+      errorWithQuickAction,
+      this.actionName
+    ),
+    actionIsUnavailable: I18nUtils.format(actionIsUnavailable, this.actionName),
+  };
+
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The result to post to the chatter feed.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * The label displayed in the tooltip of the result action button.
+   * @api
+   * @type {string}
+   */
+  @api label = this.labels.postToFeed;
+  /**
+   * The template used to generate the text to insert in the body of the chatter post.
+   * Use references to result [fields](https://docs.coveo.com/en/2036/index-content/about-fields) to get their value.
+   * @api
+   * @type {string}
+   */
+  @api textTemplate =
+    '<a href="${clickUri}">${title}</a><br/><br/><quote>${excerpt}</quote>';
+
+  /** @type {string} */
+  iconName = 'utility:share_post';
+  /** @type {string} */
+  eventName = 'insightpanel__posttofeed';
+  /** @type {string} */
+  insertType = 'cursor';
+  /** @type {boolean} */
+  _isLoading = false;
+  /** @type {boolean} */
+  _actionHandled = false;
+  /** @type {Function} */
+  unsubscribe;
+  /** @type {CoveoHeadlessInsight} */
+  headless;
+  /** @type {NodeJS.Timeout} */
+  timeout;
+  engine;
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+    this.addEventListener(this.eventName, this.handlePostToFeedClick);
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener(this.eventName, this.handlePostToFeedClick);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  initialize = (engine) => {
+    this.headless = getHeadlessBundle(this.engineId);
+    this.engine = engine;
+
+    this.actions = {
+      ...this.headless.loadInsightAnalyticsActions(engine),
+    };
+  };
+
+  postResultToFeed() {
+    this._isLoading = true;
+    this._actionHandled = false;
+
+    let resultPromiseResolve;
+    let resultPromiseReject;
+    const resultPromise = new Promise((resolve, reject) => {
+      resultPromiseResolve = resolve;
+      resultPromiseReject = reject;
+    });
+    resultPromise.catch(this.handleResultPromiseFailure).finally(() => {
+      clearTimeout(this.timeout);
+      this._actionHandled = true;
+      this._isLoading = false;
+    });
+
+    // eslint-disable-next-line @lwc/lwc/no-async-operation
+    this.timeout = setTimeout(() => {
+      if (!this._actionHandled) {
+        resultPromiseReject({unavailableAction: true});
+      }
+    }, 2000);
```

Comment:
I think since we're exposing this publicly we should explain why we have this timeout?